### PR TITLE
Add random test for strformt::fmt()

### PR DESCRIPTION
**Pull request by h3ndrk**
This adds a simple random test which tests if the function `strformat::fmt()` correctly emplaces strings into a format string. See the comment of the test for more information.

### CODE REVIEW

File: Backend/sub_crates/str_util/src/tests/strformat.rs [hunk 1]
```diff
@@ -1,11 +1,37 @@
 extern crate proptest;
-use crate::strformat;
 use self::proptest::prelude::*;
+use crate::strformat;
+
+proptest! {
+  /// This test is supplied with three random inputs:
+  ///
+  /// 1. Size of the generated arguments, in interval `[0,32)`
```
Comment (m-dobler) [inline Backend/sub_crates/str_util/src/tests/strformat.rs:8]: Small detail: Isn't the 32 included as a possible size in the argument generation `size in 0usize..33usize`?

File: Backend/sub_crates/str_util/src/tests/strformat.rs [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+  fn test_arguments(size in 0usize..33usize, permutations in prop::array::uniform32(0usize..32usize), strings in prop::array::uniform32("\\PC*")) {
+    let format = (0..size)
+      .map(|i| format!("{{{}}}", permutations[i]))
+      .collect::<Vec<_>>()
+      .join(" ");
+    let arguments = strings
+      .iter()
+      .map(|item| &item[..])
+      .collect::<Vec<_>>();
+    let expected = (0..size)
```
Comment (m-dobler) [inline Backend/sub_crates/str_util/src/tests/strformat.rs:23]: Just for my understanding: The test generates random inputs and checks whether the `fmt` function that is being tested returns the same result as "expected", where the "expected" result is being computed by a custom implementation here in the test, right? 

In general I am unsure about this type of test, because effectively, you are only testing whether these two implementations always return the same result. While it is very unlikely, the two implementations could return the same wrong results and the test wouldn't catch anything. 

I also feel like if you are so confident that you are computing the expected results correctly, you could also just implement the `fmt` function in the same way. WDYT?

File: Backend/sub_crates/str_util/src/tests/strformat.rs (hunk 1 above)
Comment (h3ndrk) [reply Backend/sub_crates/str_util/src/tests/strformat.rs]: Fixed

File: Backend/sub_crates/str_util/src/tests/strformat.rs (hunk 2 above)
Comment (h3ndrk) [reply Backend/sub_crates/str_util/src/tests/strformat.rs:23]: Your understanding is correct.

It is true that I chose a very simple function to test for which the testing possibilities are limited. I selected this function to have a gentle learning curve (because of near deadline, many new techniques, etc.). Essentially, I'm testing that the `fmt()` function correctly emplaces strings into the format string (which is the reason for the existence of this function, in my understanding).

For random testing with *proptest*, I haven't found any other method for random testing this function. If I would think like *Randoop*: I have this function and need to generate a random test that generates valid inputs. This is what I tried to do and since the output of `fmt()` is not trivial (e.g. simple boolean flag), it needs the `expected` computation.

Furthermore, in Rust we do not have as many corner cases like in Java where e.g. `null` exists. This reduces the possibilities for writing tests even more.
